src/core/binding.c

```diff
@@ -1344,7 +1345,40 @@ QuicBindingReceive(
     // connection it was delivered to.
     //
 
-    QUIC_RECV_DATAGRAM* Datagram;
+#if DEBUG
+    if (MsQuicLib.TestDatapathFuncTable != NULL) {
+
+        //
+        // The test datapath receive callback allows for test code to modify
+        // the datagrams on the receive path, and optionally indicate one or
+        // more to be dropped.
+        //
+
+        MsQuicLib.TestDatapathFuncTable->Receive(DatagramChain);

```

Comment:
One at a time rather than in chains?

Reply:
Why? I just added some example test code that uses this interface to add a random loss test case. It simply iterates through each item in the chain.

Reply:
Yeah, doesn't sound like a big deal; the point would be that you write a loop in one place (here) rather than in every interface client. And I can't think of a good reason (perf doesn't matter here) to indicate chains, so it seems like a pure win.

Reply:
(plus, you already wrote a loop here :) )

Reply:
Your logic is sound. I'll refactor it.